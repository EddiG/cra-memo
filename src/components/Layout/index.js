// @flow
import React from 'react';
import styled from 'styled-components';

type Props = {
  children?: React.Children,
};

const Container = styled.div`
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
`;

const Header = styled.header`
  display: flex;
  justify-content: center;
  min-height: 10vh;
  background-color: ${props => props.theme.colors.bg.nav};
`;

const Title = styled.h1`
  text-transform: uppercase;
  color: ${props => props.theme.colors.fg.main};
`;

const Main = styled.main`
  /* padding: 20px; */
`;

const Layout = ({ children }: Props) => (
  <Container>
    <Header>
      <Title>
        Memo
      </Title>
    </Header>
    <Main>
      {children}
    </Main>
  </Container>
);

Layout.defaultProps = {
  children: null,
};

export default Layout;
