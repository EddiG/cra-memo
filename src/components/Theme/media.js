import { css } from 'styled-components';

const baseFontSize = 16;

const sizes = {
  desktop: 1024,
  tablet: 768,
  phone: 376,
};

export default Object.keys(sizes).reduce((acc, target) => {
  const emSize = sizes[target] / baseFontSize;
  acc[target] = (...args) => css`
    @media (max-width: ${emSize}em) {
      ${css(...args)}
    }
  `;
  return acc;
}, {});
