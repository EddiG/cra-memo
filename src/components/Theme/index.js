// @flow
import React from 'react';
import { ThemeProvider } from 'styled-components';
import media from './media';

type Props = {
  children?: React.Children,
};

const theme = {
  colors: {
    fg: {
      main: '#b1b1b1',
      avatar: '#53bdff',
      button: '#fff',
    },
    bg: {
      nav: '#333333',
      main: '#fff',
      button: '#e0b305',
    },
  },
  media,
};

const Theme = ({ children }: Props) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
);

Theme.defaultProps = {
  children: null,
};

export default Theme;
