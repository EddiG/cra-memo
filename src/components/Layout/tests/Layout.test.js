import React from 'react';
import renderer from 'react-test-renderer';
import Layout from '../index';
import Theme from '../../Theme';

describe('Layout component', () => {
  it('should rendered successfully', () => {
    const layout = renderer.create(<Theme><Layout /></Theme>).toJSON();
    expect(layout).toMatchSnapshot();
  });
});
