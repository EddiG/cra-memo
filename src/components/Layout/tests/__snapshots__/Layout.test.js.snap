// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should rendered successfully 1`] = `
<div
  className="sc-bdVaJa eUYAYB"
>
  <header
    className="sc-bwzfXH cgfKFj"
  >
    <h1
      className="sc-htpNat IPSgH"
    >
      Memo
    </h1>
  </header>
  <main
    className="sc-bxivhb ODRvI"
  />
</div>
`;
